### ISSUE TITLE
[Bug]: [CI]Search failed reporting segment lacks and channel not available

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#28353]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3-20231110-f2e972ab
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    rocksmq
- SDK version(e.g. pymilvus v2.0.0rc2): 2.3.2.post1.dev9
- OS(Ubuntu or CentOS):  
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed reporting segment lacks and channel not available.
```
[2023-11-10T12:04:39.305Z] [2023-11-10 11:36:50 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.9105234800356374, 0.5301417207586637, 0.7584130783271936, 0.27463223383857605, 0.04085273758613539, 0.8180724525594242, 0.7448851132530023, 0.8454275253620324, 0.9039594901406349, 0.010358342726193959, 0.19934898458473038, 0.8353481314816271, 0.5935830465488559, 0.1577581748032708, 0.1754712930......, kwargs: {'_async': False} (api_request.py:62)
[2023-11-10T12:04:39.305Z] [2023-11-10 11:36:50 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=503, message=failed to search: segment=445544038041816278: segment lacks: channel=by-dev-rootcoord-dml_12_445544038041816194v0: channel not available)>, <Time:{'RPC start': '2023-11-10 11:36:50.452241', 'RPC error': '2023-11-10 11:36:50.456603'}> (decorators.py:128)
[2023-11-10T12:04:39.305Z] [2023-11-10 11:36:50 - ERROR - ci_test]: Traceback (most recent call last):
[2023-11-10T12:04:39.305Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper
[2023-11-10T12:04:39.305Z]     res = func(*args, **_kwargs)
[2023-11-10T12:04:39.305Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request
[2023-11-10T12:04:39.305Z]     return func(*arg, **kwargs)
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 783, in search
[2023-11-10T12:04:39.305Z]     resp = conn.search(
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 129, in handler
[2023-11-10T12:04:39.305Z]     raise e from e
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 125, in handler
[2023-11-10T12:04:39.305Z]     return func(*args, **kwargs)
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 164, in handler
[2023-11-10T12:04:39.305Z]     return func(self, *args, **kwargs)
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 104, in handler
[2023-11-10T12:04:39.305Z]     raise e from e
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 68, in handler
[2023-11-10T12:04:39.305Z]     return func(*args, **kwargs)
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 781, in search
[2023-11-10T12:04:39.305Z]     return self._execute_search(request, timeout, round_decimal=round_decimal, **kwargs)
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 742, in _execute_search
[2023-11-10T12:04:39.305Z]     raise e from e
[2023-11-10T12:04:39.305Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 734, in _execute_search
[2023-11-10T12:04:39.305Z]     raise MilvusException(response.status.code, response.status.reason)
[2023-11-10T12:04:39.305Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=503, message=failed to search: segment=445544038041816278: segment lacks: channel=by-dev-rootcoord-dml_12_445544038041816194v0: channel not available)>
[2023-11-10T12:04:39.305Z]  (api_request.py:45)
[2023-11-10T12:04:39.306Z] [2023-11-10 11:36:50 - ERROR - ci_test]: (api_response) : <MilvusException: (code=503, message=failed to search: segment=445544038041816278: segment lacks: channel=by-dev-rootcoord-dml_12_445544038041816194v0: channel not available)> (api_request.py:46)
```

### Expected Behavior

pass

### Milvus Log

https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-28344/5/pipeline

**Comment by NicoYuan1986** [comment]
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-28455/1/pipeline/141
I encounter this problem quite frequently. 
@weiliu1031 help.

**Comment by NicoYuan1986** [comment]
not reproduced for a month. master-20231218-95ada1b6

**Comment by RakeshRaj97** [comment]
I see this issue. Any updates?

**Comment by xiaofan-luan** [comment]
> I see this issue. Any updates?

it should be fixed by latest 2.3.18 or 2.4.5

**Comment by xiaofan-luan** [comment]
> I see this issue. Any updates?

please offer logs and open another issues if it's not recovered after upgrade
